subprojects/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/AbstractPropertyUpgradesBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.integtests
+
+
+import org.gradle.integtests.fixtures.CrossVersionIntegrationSpec
+
+/**
+ * Tests that task classes with property upgrades are compiled against one version of Gradle are compatible with another version.
+ */
+abstract class AbstractPropertyUpgradesBinaryCompatibilityCrossVersionSpec extends CrossVersionIntegrationSpec {
+
+    protected List<Class<?>> importClasses() {
+        return []
+    }
+
+    protected void prepareGroovyPluginTest(String pluginApplyBody) {
+        file("producer/build.gradle") << """
+            apply plugin: 'groovy'
+            dependencies {
+                implementation localGroovy()
+                implementation gradleApi()
```

Comment:
❔ Do we need both of these? I thought depending on `gradleApi()` adds `localGroovy()` automatically.